arquillian-chameleon-runner/runner/src/main/java/org/arquillian/container/chameleon/runner/ArquillianChameleon.java

```diff
@@ -0,0 +1,74 @@
+package org.arquillian.container.chameleon.runner;
+
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.logging.Logger;
+import org.arquillian.container.chameleon.runner.extension.ChameleonRunnerAppender;
+import org.jboss.arquillian.junit.Arquillian;
+import org.junit.runner.notification.RunNotifier;
+import org.junit.runners.model.InitializationError;
+
+public class ArquillianChameleon extends Arquillian {
+
+    private static final Logger log = Logger.getLogger(ArquillianChameleon.class.getName());
+
+    public ArquillianChameleon(Class<?> testClass) throws InitializationError {
+        super(testClass);
+    }
+
+    @Override
+    public void run(RunNotifier notifier) {
+
+        Class<?> testClass = getTestClass().getJavaClass();
+
+        final ClassLoader parent = Thread.currentThread().getContextClassLoader();
+
+        if (isInClientSide(parent) && !isSpecialChameleonFile(parent)) {
+
+            try {
+                Path arquillianChameleonConfiguration = new ArquillianChameleonConfigurator().setup(testClass, parent);
+
+                log.info(String.format("Arquillian Configuration created by Chameleon runner is placed at %s.", arquillianChameleonConfiguration.toFile().getAbsolutePath()));
+
+                createSpecialChameleonFile(arquillianChameleonConfiguration.getParent());
+                addsArquillianFile(arquillianChameleonConfiguration.getParent(),
+                    parent);
+            } catch (Exception e) {
+                throw new IllegalArgumentException(e);
+            }
+        }
+        super.run(notifier);
+    }
+
+    /**
+     * We need to create an special file and put it inside classloader. This is because Chameleon runner adds configuration files dynamically inside the classpath.
```

Comment:
`a special file`. not sure if that needs to be javadoc, not a regular comment.

Reply:
I have changed to a regular comment. I think that this method is really strange and in 6 months is the typical thing that we will not remember why we did it.